src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  if (dateMap['YYYY'] && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(2);
+  } else if (dateMap['YY'] && toFormat.includes('YYYY')) {
+    dateMap['YYYY'] =
+      dateMap['YY'] < 30 ? '20' + dateMap['YY'] : '19' + dateMap['YY'];
+  }
+
+  const newDate = toFormat
```

Comment:
The mapping from `dateMap` to the new date format assumes that all required parts are present in `dateMap`. Consider adding a check to ensure that each required part of `toFormat` is available in `dateMap` before attempting to map it to avoid undefined values in the output.